src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Collections/SegmentedList`1.cs

```diff
@@ -0,0 +1,794 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using IEnumerable = System.Collections.IEnumerable;
+using IEnumerator = System.Collections.IEnumerator;
+
+namespace Microsoft.CodeAnalysis.Shared.Collections
+{
+    internal interface IByRefComparer<T>
+    {
+        int Compare(in T first, in T second);
+    }
+
+    /// <summary>
+    /// Segmented list implementation, copied from Microsoft.Exchange.Collections.
```

Comment:
safe to license as per dotnet/MIT license?

Reply:
I'm assuming yes since we put it on PerfView already.